pkg/controller/atlascluster/cluster.go

```diff
@@ -130,3 +132,37 @@ func clustersEqual(log *zap.SugaredLogger, clusterA mongodbatlas.Cluster, cluste
 
 	return d == ""
 }
+
+func ensureConnectionSecrets(ctx context.Context, wctx *workflow.Context, k8sClient client.Client, project *mdbv1.AtlasProject, cluster *mongodbatlas.Cluster) workflow.Result {
+	dbUsers, _, err := wctx.Client.DatabaseUsers.List(ctx, project.ID(), &mongodbatlas.ListOptions{})
```

Comment:
we need to query K8s database users as we know the passwords for them only - not for Atlas db users

Reply:
Ah, that makes sense... Guess this is going to be a bit more complicated then :slightly_smiling_face: